gcovr/writer/json.py

```diff
@@ -248,6 +250,19 @@ def _json_from_branch(branch):
     return json_branch
 
 
+def _json_from_decision(decisions):
+    json_decision = {}
+    if len(decisions) == 1:
+        json_decision["is_switch"] = True
+        json_decision["count_true"] = decisions[0].count
+    elif len(decisions) == 2:
+        json_decision["is_switch"] = False
+        json_decision["count_true"] = decisions[0].count
+        json_decision["count_false"] = decisions[1].count
+        json_decision["uncheckable"] = decisions[0].uncheckable
```

Comment:
Is it guaranteed that there will be exactly 0, 1, or 2 decisions on the line? Because the JSON format MUST round-trip losslessly. A data format like this that interprets the decision as a “switch” might be more immediately useful, but it would greatly limit how the JSON format can be used by gcovr itself, for example to merge coverage data from multiple test runs.

Reply:
If there are 0 decisions the function isn't called. I think a key/value would be better than using an index and a representation in JSON like e.g.:
```json
{
   "switch" : { "count_true": 3 }
}
```
```json
{
    "count_false": 1,
    "count_true": 3,
    "uncheckable": false
}
```

Reply:
I think the current JSON data model is better than the `DecisionCoverage` data model at representing the result of the decision coverage analysis. Indeed, using indexes to disambiguate the true/false decisions is confusing.

I'd leave the JSON format as it is, or use a more extensible representation like

```json
{
  "type": "switch",
  "count": 3
}
```

```json
{
  "type": "conditional",
  "count_true": 1,
  "count_false": 3,
}
```

```json
{
  "type": "uncheckable"
}
```

Whether using a representation like `{ "switch": {...} }` or `{ "type": "switch", ... }` is mostly a matter of style. The former is more human-readable, the latter is typically easier to handle in software. I have no strong feelings about this.

Reply:
I'll improve the JSON data model with the type, this is easier to understand. I also check if this data model can be applied to `DecisionCoverage`.

Reply:
I've added seperate coverage classes and changed the JSON using the `type` key.